### PR TITLE
Integratie frontend

### PR DESCRIPTION
**Pull request by Badlapje**

### CODE REVIEW

File: src/Surfnet/ServiceProviderDashboard/Infrastructure/HttpClient/HttpClient.php [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+    private $apiName;
+
+    /**
+     * @param string $mode The mode is used mainly for logging purposes, stating which environment was targeted.
+     */
+    public function __construct(
+        ClientInterface $httpClient,
+        LoggerInterface $logger,
+        string $apiName = 'testApi',
+        string $mode = self::MODE_TEST
```
Comment (MKodde) [inline src/Surfnet/ServiceProviderDashboard/Infrastructure/HttpClient/HttpClient.php:64]: Why is the one set in `const` where the other is not?

File: src/Surfnet/ServiceProviderDashboard/Infrastructure/Teams/Client/QueryClient.php [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+/**
+ * The QueryClient can be used to perform queries on the manage /api/spdashboard/teams/<urn> endpoint
+ *
+ * Queries will return the full team information.
+ *
+ * Example response (json formatted for readability)
+ *  {
+        "id": 1,
+        "urn": "demo:openconext:org:riders",
+        "name": "riders",
```
Comment (MKodde) [inline src/Surfnet/ServiceProviderDashboard/Infrastructure/Teams/Client/QueryClient.php:38]: I love the example response and the description of the class usage. :heart:

File: src/Surfnet/ServiceProviderDashboard/Application/CommandHandler/Service/CreateServiceCommandHandler.php [hunk 3]
```diff
@@ -59,9 +96,35 @@ public function handle(CreateServiceCommand $command)
...
+            'description' => $this->translator->trans('teams.create.description', [
+                '%teamName%' => $name
+            ]),
+            'personalNote' => $this->translator->trans('teams.create.personalNote'),
+            'viewable' => false,
+            'emails' => $emails,
+            'roleOfCurrentUser' => 'MANAGER',
+            'invitationMessage' => $this->translator->trans('teams.create.invitationMessage'),
+            'language' => 'ENGLISH',
+        ];
```
Comment (MKodde) [inline src/Surfnet/ServiceProviderDashboard/Application/CommandHandler/Service/CreateServiceCommandHandler.php:118]: Like in my previous Profile review. I'd love to see use of a DTO or Value object like solution for this kind of data transfer.

File: src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Controller/TeamsController.php [hunk 4]
```diff
@@ -143,6 +163,35 @@ public function manageTeamAction(Request $request, int $serviceId)
...
+        $this->get('session')->getFlashBag()->clear();
+        $email = $request->get('email');
+        $role = $request->get('role');
+        $invite = [
+            'teamId' => $teamId,
+            'intendedRole' => $role,
+            'emails' => $this->createEmailsArray($email, $role),
+            'message' => $this->translator->trans('teams.create.invitationMessage'),
+            'language' => 'ENGLISH',
+        ];
```
Comment (MKodde) [inline src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Controller/TeamsController.php:132]: Again, consider changing this into a more testable and dependable solution using VO's or DTO's

**Review by MKodde — COMMENTED** [review]
Very nice work (and lots of it). It was not easy to review everything with the attention it deserves, and I do not claim to have seen all possible quirks. I focussed my attention on the newly added files. And other eye catchers. See some findings below.

Another point I noticed was the somewhat meager test coverage of the new logic. Some tests I expected to see where unit tests for the new classes and a couple of web tests covering the new logic. I find these two test types essential and should be created to be able to support your changes in the future.

An other finding is that you stepped away from the command handling based solution when it comes to crudding teams and team-memberships. I see you chose to go a more pragmatic TypeScript based solution. And I can certainly live with that. I'd love to hear more background on this. Why did you choose to go that way over the current standard. It might deserve a little writeup in a readme. And another thing is, I do not see TS test coverage. The business logic caught in that code deserves tests of their own.

File: src/Surfnet/ServiceProviderDashboard/Domain/Repository/DeleteTeamsEntityRepository.php [hunk 5]
```diff
@@ -0,0 +1,34 @@
+<?php
+
+/**
+ * Copyright 2018 SURFnet B.V.
```
[the remark this answers is not in the document]
Comment (Badlapje) [reply src/Surfnet/ServiceProviderDashboard/Domain/Repository/DeleteTeamsEntityRepository.php:4]: fixed

File: src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Controller/SiteNoticeController.php [hunk 6]
```diff
@@ -45,7 +45,6 @@ public function __construct(string $noticeDate)
      * session and in cookies.
      *
      * @Method("GET")
-     * @Security("has_role('ROLE_USER')")
```
[the remark this answers is not in the document]
Comment (Badlapje) [reply src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Controller/SiteNoticeController.php:48]: It gave errors in some situations due to the way how it is called.  I searched for a while on how to resolve it without doing this, but couldn't find anything.

File: src/Surfnet/ServiceProviderDashboard/Infrastructure/HttpClient/HttpClient.php (hunk 1 above)
Comment (Badlapje) [reply src/Surfnet/ServiceProviderDashboard/Infrastructure/HttpClient/HttpClient.php:64]: good question, adjusted this

File: tests/unit/Infrastructure/Manage/Client/PublishEntityClientTest.php [hunk 7]
```diff
@@ -16,7 +16,7 @@
  * limitations under the License.
  */
 
-namespace Surfnet\ServiceProviderDashboard\Tests\Unit\Infrastructure\DashboardBundle\Service;
+namespace Infrastructure\Manage\Client;
```
Comment (MKodde) [inline tests/unit/Infrastructure/Manage/Client/PublishEntityClientTest.php:19]: The vendor and application seems to be missing from the namespace here

File: tests/unit/Infrastructure/Teams/Client/PublishEntityClientTest.php [hunk 8]
```diff
@@ -0,0 +1,136 @@
+<?php
+
+/** Copyright 2021 SURFnet B.V.
```
Comment (MKodde) [inline tests/unit/Infrastructure/Teams/Client/PublishEntityClientTest.php:3]: The Copyright line must be on the next line (4). That fixes your static-analysis issue (the docheader at least)

Comment (Badlapje) [reply tests/unit/Infrastructure/Teams/Client/PublishEntityClientTest.php:3]: superb, much obliged!

**Pull request merged**